### DISCUSSION TITLE
Is there a way of getting snippet completion with the native vim completion?

### DISCUSSION THREAD

**Discussion by Supreeeme** [L3MON4D3/LuaSnip#615] [Q&A] [answered]
I've been trying out the built in vim completion with `vim.lsp.omnifunc` set to my omnifunc, and I've liked it a lot. My only issue I have with it is I don't seem to have a way of getting snippet completion. Is there something I'm missing here, is this possible?

**Comment by L3MON4D3** [comment, marked as answer]
We don't yet support omnifunc, sorry :/

If you want to implement it though, you can extract relevant code (get list of snippets, expand chosen snippet) from `cmp_luasnip`.
I can't give any advice on writing a new source for omnifunc, but I'm certain that there is a source for ultisnips/vim-vsnip/snipmate out there, which might be adaptable.
(If you end up implementing it, definitely add it to the wiki, or the readme :D)

**Reply by Supreeeme** [reply]
Dang, that's unfortunate. I'll see if I can whip something up.
